extensions/ui/org.eclipse.smarthome.ui.paper/web-src/js/rules/controllers.rules.js

```diff
@@ -295,6 +296,173 @@ angular.module('PaperUI.controllers.rules', [ 'PaperUI.controllers.extension' ])
     $scope.close = function() {
         $mdDialog.hide();
     }
+}).controller('addModuleDialogController', function($rootScope, $scope, $mdDialog, moduleTypeService, sharedProperties, $filter, configService, module, ruleID, type) {
+
+    var objectFilter = $filter('filter');
+    $scope.moduleData;
+    $scope.triggerData;
+
+    getModulesByType(type, type == 'trigger' ? setConfigurations : null);
+
+    function getModulesByType(mtype, callback) {
+        moduleTypeService.getByType({
+            mtype : mtype
+        }).$promise.then(function(data) {
+            var modules = objectFilter(data, {
+                visibility : 'VISIBLE'
+            });
+            if (mtype != 'trigger' || type == 'trigger') {
+                $scope.moduleData = modules;
+            }
+            if (callback) {
+                $scope.triggerData = modules;
+                if ($scope.module) {
+                    callback();
+                }
+            } else {
+                getModulesByType('trigger', setConfigurations);
+            }
+
+        });
+    }
+    $scope.id = module.id;
+    $scope.type = type;
+    $scope.description = '';
+    $scope.module = '';
+    $scope.step = 1;
+    $scope.editMode = false;
+    $scope.configuration = {};
+    $scope.parameters = [];
+    var originalConfiguration = {};
+    $scope.items = [];
+
+    $scope.selectChip = function(chip, textAreaName) {
+        var textArea = $("textarea[name=" + textAreaName + "]")[0];
```

Comment:
Is ES6 allowed here? If so you could use a template string.
Also, it's a good practice to include quotes in pseudoselectors.
```
$(`textarea[name="${textAreaName}"]`)[0];
```